AttributeError: 'Text' object has no attribute 'enabled'

@johnsca hit this one today, not sure if this is related to some of the keyboard navigation changes?

```
2018-02-01 15:24:43,875 [DEBUG] conjure-up/_unspecified_spell - juju.py:39 - bin_path candidate found
2018-02-01 15:24:43,875 [DEBUG] conjure-up/_unspecified_spell - juju.py:39 - wait_path candidate found
2018-02-01 15:24:43,975 [DEBUG] conjure-up/_unspecified_spell - app.py:254 - Juju version: 2.3.1-xenial-amd64, conjure-up version: 2.5.3-rc1
2018-02-01 15:24:43,992 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:17 - Showing screen: Application Start
2018-02-01 15:24:43,997 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:31 - OS: Linux-4.4.0-1047-aws-x86_64-with-debian-stretch-sid 
2018-02-01 15:24:44,000 [INFO] conjure-up/_unspecified_spell - events.py:172 - Watching for shutdown
2018-02-01 15:24:44,038 [DEBUG] conjure-up/_unspecified_spell - events.py:53 - Awaiting Shutdown at conjureup/events.py:175
2018-02-01 15:24:44,051 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:17 - Showing screen: Spell Selection
2018-02-01 15:24:48,801 [DEBUG] conjure-up/_unspecified_spell - telemetry.py:31 - Spell Choice: openstack-novalxd 
2018-02-01 15:24:48,803 [DEBUG] conjure-up/openstack-novalxd - download.py:60 - Path is local filesystem, copying /snap/conjure-up/947/spells/openstack-novalxd to /home/ubuntu/.cache/conjure-up/openstack-novalxd
2018-02-01 15:24:48,945 [DEBUG] conjure-up/openstack-novalxd - step.py:27 - steps: [<StepModel OpenStack with NovaLXD 00_process-providertype v: False c: []>, <StepModel OpenStack with NovaLXD 01_glance v: True c: []>, <StepModel OpenStack with NovaLXD 02_keypair v: True c: []>, <StepModel OpenStack with NovaLXD 03_neutron v: True c: []>, <StepModel OpenStack with NovaLXD 04_horizon v: True c: []>]
2018-02-01 15:24:49,609 [DEBUG] conjure-up/openstack-novalxd - events.py:53 - Setting Error at conjureup/events.py:148
2018-02-01 15:24:49,612 [ERROR] conjure-up/openstack-novalxd - events.py:160 - Unhandled exception
Traceback (most recent call last):
  File "/snap/conjure-up/947/usr/lib/python3.6/asyncio/events.py", line 127, in _run
    self._callback(*self._args)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/raw_display.py", line 393, in <lambda>
    event_loop, callback, self.get_available_raw_input())
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/raw_display.py", line 493, in parse_input
    callback(processed, processed_codes)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/main_loop.py", line 403, in _update
    self.process_input(keys)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/main_loop.py", line 503, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/wimp.py", line 643, in keypress
    return self._current_widget.keypress(size, key)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/urwid/container.py", line 1128, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/base.py", line 391, in keypress
    result = self._command_handlers[command]()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/base.py", line 346, in submit_field
    self.submit()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/base.py", line 364, in submit
    self.next_screen()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/spellpicker.py", line 72, in next_screen
    self.cb(self.selected_spell['key'])
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/controllers/spellpicker/gui.py", line 25, in finish
    return controllers.use('addons').render()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/controllers/addons/gui.py", line 15, in render
    return self.next_screen()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/controllers/addons/gui.py", line 33, in next_screen
    controllers.use('clouds').render()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/controllers/clouds/gui.py", line 76, in render
    back=prev_screen)
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/cloud.py", line 42, in __init__
    self.update_message()
  File "/snap/conjure-up/947/lib/python3.6/site-packages/conjureup/ui/views/cloud.py", line 49, in update_message
    if selected.enabled:
AttributeError: 'Text' object has no attribute 'enabled'
2018-02-01 15:24:49,621 [DEBUG] conjure-up/openstack-novalxd - __init__.py:29 - Showing dialog for exception: 'Text' object has no attribute 'enabled'
2018-02-01 15:24:53,596 [DEBUG] conjure-up/openstack-novalxd - events.py:53 - Setting Shutdown at conjureup/events.py:144
2018-02-01 15:24:53,597 [DEBUG] conjure-up/openstack-novalxd - events.py:53 - Received Shutdown at conjureup/events.py:175
2018-02-01 15:24:53,597 [INFO] conjure-up/openstack-novalxd - events.py:179 - Shutting down
```

--- a/conjureup/ui/views/cloud.py
+++ b/conjureup/ui/views/cloud.py
@@ -45,8 +45,10 @@ class CloudView(BaseView):
         self.update_message()
 
     def update_message(self):
-        selected = self.widget.focus
-        if selected.enabled:
+        selected = self.widget.selected_widgets
+        if selected is None:
+            return
+        elif selected.enabled:
             msg = self.enabled_msg
         else:
             msg = selected.user_data.get('disabled_msg',
@@ -59,6 +61,8 @@ class CloudView(BaseView):
         if self._items_localhost_idx is None:
             return
         self.widget.contents[self._items_localhost_idx][0].enabled = True
+        if self.widget.selected_widgets is None:
+            self.widget.select_first()
         self.update_message()
 
     def build_widget(self):

--- a/conjureup/ui/widgets/selectors.py
+++ b/conjureup/ui/widgets/selectors.py
@@ -116,22 +116,34 @@ class SelectList(Pile):
                 if isinstance(item[0], self.option_type)]
 
     @property
-    def selected(self):
-        """
-        Get the values of all selected option items.
-        """
+    def selected_widgets(self):
         if not hasattr(self.option_widgets[0], 'state'):
-            if getattr(self.focus, 'enabled', True):
-                return self.focus.value
+            widget = self.focus
+            is_opt = isinstance(widget, self.option_type)
+            if is_opt:
+                return widget
             else:
                 return None
         else:
-            selected = [option.value for option in self.option_widgets
+            selected = [option for option in self.option_widgets
                         if option.state and getattr(option, 'enabled', True)]
             if self.allow_multiple:
                 return selected
             else:
                 return selected[0] if selected else None
+
+    @property
+    def selected(self):
+        """
+        Get the values of all selected option items.
+        """
+        widgets = self.selected_widgets
+        if isinstance(widgets, list):
+            return [w.value for w in widgets]
+        elif getattr(widgets, 'enabled', True):
+            return widgets.value
+        else:
+            return None
 
     @property
     def value(self):

